src/snowflake/snowpark/session.py

```diff
@@ -460,6 +460,7 @@ def __init__(
                 _PYTHON_SNOWPARK_USE_SQL_SIMPLIFIER_STRING, True
             )
         )
+        self._cte_optimization_enabled: bool = False
```

Comment:
overall I don't find any issues with the change. can we set this param using an environment variable and add a merge gate to integ tests with this param enabled?

Reply:
Maybe we follow the similar step when we added SQL simplifier? Add a server side parameter and add a merge gate. And maybe I can do it in next PR (integrate CTE optimization with SQL simplifier).

Reply:
sounds good

Reply:
@sfc-gh-yixie wdyt?

Reply:
Agree